### DISCUSSION TITLE
Weird pendingMs behavior

### DISCUSSION THREAD

**Discussion by GabriFila** [TanStack/router#243] [Q&A] [answered]
Hi everyone, loving react-location, I use it for its url-search-related features.
I added code-route-splitting. I wanted to add/play with the `pendingMs `and `pendingMinMs` settings.

**_I've prepared a sandbox with an example_**
https://codesandbox.io/s/busy-frog-lr9x4p?file=/src/App.tsx

It is a simple application with 3 dynamically loaded routes:
- homepage on `/`
- route A on `/a`
- route B on `/b`

I set these defaults in my router:
- defaultPendingMinMs=100000
- defaultPendingMs=1
- defaultPendingElement=(a fullscreen red div)

If I understand correctly react-location, this settings should tell the router to show the pending element after 1ms (so almost always) of waiting and show it for 100s, then show the route.

But this behavior doesn't show, meaning that when I change a route, the outlet is blank for a split second (or more if I slow the network through devtools) and then it renders.

Could you please explain what I am missing here?

Thank you!

**Comment by tannerlinsley** [comment, marked as answer]
It looks like a bug. In this example: https://codesandbox.io/s/reverent-leakey-tyu99d?file=/src/App.tsx, the *loader* does trigger the pending state, but not the element. I'll look into this.
